xbmc/platform/win32/WIN32Util.cpp

```diff
@@ -1474,28 +1456,60 @@ HDR_STATUS CWIN32Util::GetWindowsHDRStatus()
       displayManager.Close();
     }
   }
+
+  if (hdrSupported)
+    status = hdrEnabled ? HDR_STATUS::HDR_ON : HDR_STATUS::HDR_OFF;
+
 #else
-  uint32_t pathCount = 0;
-  uint32_t modeCount = 0;
 
-  MONITORINFOEXW mi = {};
-  mi.cbSize = sizeof(mi);
-  GetMonitorInfoW(MonitorFromWindow(g_hWnd, MONITOR_DEFAULTTOPRIMARY), &mi);
-  const std::wstring deviceNameW = mi.szDevice;
+  if (g_hWnd != NULL)
+  {
+    std::wstring gdiDeviceName{GetCurrentDisplayName()};
+    if (gdiDeviceName.empty())
+      return status;
 
-  if (ERROR_SUCCESS != GetDisplayConfigBufferSizes(QDC_ONLY_ACTIVE_PATHS, &pathCount, &modeCount))
-    return status;
+    const auto identifier{GetDisplayTargetId(gdiDeviceName)};
+    if (identifier)
+      status = GetDisplayHDRStatus(identifier.value());
+  }
+  else
+  {
+    const CWIN32Util::DisplayConfigSnapshot snapshot{GetDisplayConfigSnapshot()};
 
-  std::vector<DISPLAYCONFIG_PATH_INFO> paths(pathCount);
-  std::vector<DISPLAYCONFIG_MODE_INFO> modes(modeCount);
+    for (const auto& path : snapshot.paths)
+    {
+      const CWIN32Util::DisplayConfigId identifier{path.targetInfo.adapterId, path.targetInfo.id};
+      HDR_STATUS temp{GetDisplayHDRStatus(identifier)};
 
-  if (ERROR_SUCCESS != QueryDisplayConfig(QDC_ONLY_ACTIVE_PATHS, &pathCount, paths.data(),
-                                          &modeCount, modes.data(), 0))
-    return status;
+      if (temp != HDR_STATUS::HDR_UNSUPPORTED)
+      {
+        status = temp;
 
-  DISPLAYCONFIG_SOURCE_DEVICE_NAME getSourceName = {};
-  getSourceName.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SOURCE_NAME;
-  getSourceName.header.size = sizeof(getSourceName);
+        if (temp == HDR_STATUS::HDR_ON)
+          break;
+      }
```

Comment:
When kodi starts and is query HDR display capability for log banner i.e:

"Display HDR capable is detected and Windows HDR switch is OFF"

It is not know yet in what display kodi will be run (window no created yet and kodi settings not read yet), then log informs of any HDR display capable and Windows HDR switch state only.

Not has "preference" if HDR is On of Off, so the loop should iterate until is HDR display is detected only. This code can be more simple without temp variable:

```c++
    for (const auto& path : GetDisplayConfigSnapshot().paths)
    {
      const DisplayConfigId identifier{path.targetInfo.adapterId, path.targetInfo.id};

      status = GetDisplayHDRStatus(identifier);

      if (status != HDR_STATUS::HDR_UNSUPPORTED)
        break;
    }
```

Reply:
The master code prefers hdr on, because it only sets hdrEnabled and hdrSupported to true when iterating, and the loop doesn't stop with the first screen that has hdr support.
I copied the behavior of master here but have no problem changing it.

I find that log message a little confusing anyway, that's a note in the PR description. In multiple screen situations, some can be SDR only, some hdr off, some hdr on and that message is not clear as a summary.

For comparison, the log line "Desktop Resolution: 1920x1200 32Bit at 60Hz" is clear.
Doing the same for HDR support would also be clear, like "Desktop is HDR capable and Windows HDR switch is off"

Other screens could be added to the banner, but as separate lines for clarity. ex:
Desktop Desktop Resolution: 1920x1200 32Bit at 60Hz, not HDR capable
Secondary screen Resolution: 3840x2160 32Bit at 23Hz, HDR capable and Windows HDR switch is off
Secondary screen Resolution: 1920x1080 32Bit at 60Hz, not HDR capable

Different PR to rework the start banner.

Reply:
This log line was only for users uploading logs, to know at a glance if the system supports HDR and if they have it enabled in Windows. Early versions of Kodi with HDR support had issues if users had HDR enabled before opening Kodi.

Yeah, this could be improved, but is not the reason of this PR.

Reply:
ok so no change then?